enterprise/frontend/src/metabase-enterprise/metabot/metabot.unit.spec.tsx

```diff
@@ -170,6 +170,37 @@ describe("metabot", () => {
 
       jest.useRealTimers();
     });
+
+    // eslint-disable-next-line jest/expect-expect
+    it("should properly auto-close metabot if route changes", async () => {
+      const { store } = setup();
+
+      // should close on normal route change
+      await assertVisible();
+      act(() => window.history?.pushState(null, "", "/1"));
+      await assertNotVisible();
+
+      // should not close on route change if metabot is processing
+      showMetabot(store.dispatch);
+      act(() => store.dispatch(setIsProcessing(true)));
+      act(() => window.history?.pushState(null, "", "/2"));
+      await assertVisible();
+      act(() => store.dispatch(setIsProcessing(false)));
+      hideMetabot(store.dispatch);
+      await assertNotVisible();
```

Comment:
why not make these blocks all separate tests? will be easier to debug which part is failing then